cinder/volume/api.py

```diff
@@ -799,9 +806,40 @@ def initialize_connection(self, context, volume, connector):
             msg = _("The volume connection cannot be initialized in "
                     "maintenance mode.")
             raise exception.InvalidVolume(reason=msg)
-        init_results = self.volume_rpcapi.initialize_connection(context,
-                                                                volume,
-                                                                connector)
+
+        def _migrate_and_initialize_connection():
+            volume_type = {}
+            if volume.volume_type_id:
+                volume_type = volume_types.get_volume_type(
+                    context.elevated(), volume.volume_type_id)
+            request_spec = {
+                'volume_properties': volume,
+                'volume_type': volume_type,
+                'volume_id': volume.id}
```

Comment:
This part seems to be the same as in `_extend()`. I wasn't sure if these are the right objects. Did you copy it from there?

Reply:
Yes, it's the same in `_extend` and in `migrate_volume`.